### PR TITLE
[READY] Added engine blocks and engine deconstruction recipes.

### LINKED ISSUE #18419: enable more parts salvageable from vehicles
Im sure this has been discussed before but i wanted to make a ticket for reference.  Just a suggestion. 

rubber hose - so many hoses on a car
spring - hood latches, suspension coils on some (though not easy to extract, possible by bashing though)
horn - (amplifier circuit)
radio - radio, amplifier circuit, wiring, etc

### PR DESCRIPTION
**Pull request by kevingranade**
Fixes #18419 
Started by @LyleSY, I mostly just fixed some syntax.
This does the bare minimum, but the various recipes need to be tweaked to yield different amounts of parts.

### CODE REVIEW

**Comment by LyleSY** [comment]
Nice, thanks!

**Comment by kevingranade** [comment]
To be clear, I'm asking for feedback on what to set the various item drop rates to.  I can either incorporate feedback here, or if someone wants to re-branch off of this and open a new PR that tweaks the items, that works too.

File: data/json/recipes/engines.json [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    "qualities": [ { "id": "SCREW", "level": 1 }, { "id": "WRENCH", "level": 1 }, { "id": "WRENCH_FINE", "level": 1 } ]
+  },
+  {
+    "type": "uncraft",
+    "result": "1cyl_combustion",
+    "copy-from": "engine_disassembly",
+    "difficulty": 4,
+    "time": 40000,
+    "components": [
+      [ [ "drivebelt", 1 ] ],
```
Comment (boydkr) [inline data/json/recipes/engines.json]: All of these recipes include `drivebelt` twice

**Comment by kevingranade** [comment]
It might be a drive belt plus an accessory belt, but if so it should be
```[ "drive_belt", 2 ]``` to avoid confusion, so it needs to change.

File: data/json/items/vehicle/engine.json [hunk 2]
```diff
@@ -161,5 +161,60 @@
     "volume": 34,
     "price": 32000,
     "displacement": 1200
+  },
+  {
+    "id": "engine_block_massive",
+    "copy-from": "spare_part",
+    "type": "GENERIC",
+    "name": "large engine block",
```
Comment (Rawng) [inline data/json/items/vehicle/engine.json]: Massive engine block is named large, and small engine block is named medium

File: data/json/recipes/engines.json [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+      [ [ "filter_air", 1 ] ],
+      [ [ "drivebelt", 1 ] ],
+      [ [ "filter_liquid", 1 ] ],
+      [ [ "motor_small", 1 ] ],
+      [ [ "well_pump", 2 ] ],
```
Comment (BorkBorkGoesTheCode) [inline data/json/recipes/engines.json]: I think adding a unique fuel pump item might be better than using a well pump for fault repair.

Comment (kevingranade) [reply data/json/recipes/engines.json]: This is outside the scope of this PR.  It requires new items, adjusting the vehicle faults, possibly crafting recipes, etc.  It's valid as its own issue though.

File: data/json/items/vehicle/engine.json [hunk 4]
```diff
@@ -161,5 +161,60 @@
...
+    "weight": 90000,
+    "volume": 10,
+    "price": 10000
+  },
+  {
+    "id": "engine_block_small",
+    "copy-from": "spare_part",
+    "type": "GENERIC",
+    "name": "small engine block",
+    "description": "The beginnings of a medium size gas or diesel engine. It's not good for much of anything on its own.",
```
Comment (Night-Pryanik) [inline data/json/items/vehicle/engine.json]: "medium" size should be changed to "small".

**Comment by snipercup** [comment]
The parts created by this PR are never spawned or referenced. for example, search engine_block_tiny.

**Comment by anothersimulacrum** [comment]
They appear in the uncraft recipes? 
image

**Comment by snipercup** [comment]
You're right. for a minute there I thought Github search was pulling my leg. I'll think about what they can be used for. An uncraft recipe for the engine_block_tiny would be a start.

**Pull request merged**